conf/config.go

```diff
@@ -288,6 +288,8 @@ func SaveConfigFile(config *MenderConfigFromFile, filename string) error {
 func maybeHTTPSClient(c *MenderConfig) *client.HttpsClient {
 	if c.HttpsClient.Certificate != "" && c.HttpsClient.Key != "" {
 		return &c.HttpsClient
+	} else if c.HttpsClient.DisableKeepAlive {
+		return &c.HttpsClient
```

Comment:
AFAICS, `HttpsClient` is only checked [here](https://github.com/mendersoftware/mender/blob/1b3b946337573e1f20ab205d8a1868c9c7bc13ee/client/client.go#L512), which is right next to the part where you add `DisableKeepAlive` to the transport. So is this really needed?

Reply:
I will check that -- but this was the first version that worked.

Reply:
yes, it is, since `newHttpsClient` gets called from `New(conf Config)` and it is not what is in the config file, but what `GetHttpConfig() client.Config` returns.

Reply:
Ok, now I see what you mean.

But this made me realize that `DisableKeepAlive` does not belong in `HttpsClient`, which is about client side HTTPS settings. `DisableKeepAlive` is equally valid for HTTP. So how about a `Connections` sub section or something like that?

```json
{
  "Connections": {
    "DisableKeepAlive": true
  }
}
```

I'm thinking that this is a place we could also put `MaxDownloadSpeed` which is another connection related requested feature.

Reply:
good idea! in the mean time I added a new option: `IdleConnTimeoutSeconds` -- I am now running tests with that (I think it is something you suggested) it may make more sense to have it as well.

Reply:
I kind of added `Connectivity` looks good?